### PR TITLE
Dkan json form

### PR DESCRIPTION
**Pull request by fmizzell**
## QA Steps
- Enable **DKAN JSON Form** module
- Add dataset at `admin/dkan/dataset`

```
rm -R ../build
npm run build
dktl drush dkan-metadata-form:sync
dktl drush cr
```

### CODE REVIEW

**Comment by janette** [comment]
- Seems that the IRI `dcat:Distribution` field should be disabled?
- Let's use an autocomplete component for Format and Media Type
- DowloadURL and AccessURL should be an either/or kind of thing
- How about a date selector on the distribution Release Date field

**Comment by janette** [comment]
Remaining tasks:
- Submit PR on react-jsonschema-form repo for accessibility issues
- Create custom validator for date-time widget to allow partial dates

File: dkan2.info.yml [hunk 1]
```diff
@@ -16,6 +16,7 @@ dependencies:
   - dkan_data
   - dkan_datastore
   - dkan_dummy_content
+  - dkan_json_form
```
Comment (fmizzell) [inline dkan2.info.yml]: Are we happy with that module name? dkan_json_form?

File: modules/custom/dkan_data/config/install/core.entity_form_display.node.data.default.yml [hunk 2]
```diff
@@ -7,10 +7,9 @@ dependencies:
     - field.field.node.data.field_json_metadata
     - node.type.data
   module:
-    - json_form
     - path
 _core:
-  default_config_hash: XUF7rshmx-Q17HcczdSSt8-HDS3hQtgY-wpX7wKI3Hg
```
Comment (fmizzell) [inline modules/custom/dkan_data/config/install/core.entity_form_display.node.data.default.yml]: lets get rid of these hashes

File: modules/custom/dkan_json_form/js/app/README.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+When ready to commit your changes
+- Remove the value from `const baseUrl = "";`
+- `rm -R ../build`
+- `npm run build`
+- `dktl drush dkan-json-form:sync`
+- `dktl drush cr`
+- Review the form at `/admin/dkan/dataset`
+
+
+## Available Scripts
```
Comment (fmizzell) [inline modules/custom/dkan_json_form/js/app/README.md]: Should we get rid of the documentation created by react app?

File: dkan2.info.yml (hunk 1 above)
Comment (janette) [reply dkan2.info.yml]: not unhappy with it... would dkan_dataset_form be better

File: modules/custom/dkan_data/config/install/core.entity_form_display.node.data.default.yml (hunk 2 above)
Comment (janette) [reply modules/custom/dkan_data/config/install/core.entity_form_display.node.data.default.yml]: ok

File: modules/custom/dkan_json_form/js/app/README.md (hunk 3 above)
Comment (janette) [reply modules/custom/dkan_json_form/js/app/README.md]: yes

**Comment by dafeder** [comment]
## Issues

1. Identifier field should be hidden and autogenerated on submit
2. When a required field is not filled out, you get a message on submit but hard to tell which field it's trying to point out
3. When there is a schema validation error, it displays nicely at the top of the page but if you're all the way at the bottom (which you woud be, because that's where the submit button is) you can't see that anything happened; feels like the submit didn't work.
4. Contact email field expects a "mailto:" string, should just need a regular email address
5. In the "datasets" admin view, the edit link works but the title links to a missing page

**Pull request merged**